functions/pm_functions.scm

```diff
@@ -435,13 +457,111 @@
 		 ))
 	 )
 
-		(EvaluationLink
-		  (PredicateNode "interacts_with")
-		   (ListLink
-			   (VariableNode "$c")
-			   (VariableNode "$b")
-		))
+  ;; This will be executed if the above pattern is found.
+  (ExecutionOutputLink
+    (GroundedSchemaNode "scm: generate-result")
+		  (ListLink
+		    (VariableNode "$c")
+		    (VariableNode "$b")
+		  ))
 	
-	))
+))
+))
+
+;; Grounded schema node to add info about matched variable nodes
+
+(define (generate-result var1 var2)
+  (if  
+    (and (not (equal? (cog-type var1) 'VariableNode)) (not (equal? (cog-type var2) 'VariableNode))) 
+
+    (begin
+    (let ([output (ListLink
+          (EvaluationLink (PredicateNode "Interacts_with") (ListLink var1 var2))
+          (node-info var2)
+          (node-info var1)
+          )])
+    output
+    )
+    )
+))
+;; build description URL of a node
+
+(define (build-desc-url node)
```

Comment:
Should this function be in this file as it isn't doing any pattern matching? Better if it is moved to `utils.scm`